grassroot-integration/src/main/java/za/org/grassroot/integration/location/LocationInfoBrokerImpl.java

```diff
@@ -367,6 +369,88 @@ public Province getProvinceFromGeoLocation(GeoLocation location) {
+            case ZA_LP:     areaId = "4292/children";       break;//uses MapIt area id
+            case ZA_NC:     areaId = "4295/children";       break;
+            case ZA_EC:     areaId = "4288/children";       break;
+            case ZA_KZN:    areaId = "4291/children";       break;
+            case ZA_FS:     areaId = "4289/children";       break;
+            case ZA_MP:     areaId = "4293/children";       break;
+            case ZA_NW:     areaId = "MDB:NW/children";     break;
+            case ZA_WC:     areaId = "MDB:WC/children";     break;
+            default:        areaId = "UNDEFINED";           break;
+        }
+
+        if(areaId.equals("UNDEFINED")){
+            return new ArrayList<>();
+        }
+        componentsBuilder = componentsBuilder.path(areaId);
+        log.info("Composed URI: {}", componentsBuilder.toUriString());
+
+        ParameterizedTypeReference<Map<String, Municipality>> responseType =
+                new ParameterizedTypeReference<Map<String, Municipality>>() {};
+
+        ResponseEntity<Map<String, Municipality>> result =
+                restTemplate.exchange(componentsBuilder.build().toUri(), HttpMethod.GET, null, responseType);
+
+        log.info("Received response: {}", result);
+
+        List<Municipality> municipalities = new ArrayList<>();
+        if(result.getBody() != null){
+           municipalities = new ArrayList<>(result.getBody().values());
+        }
+
+        log.info("Processed munis: {}", municipalities);
+
+        return municipalities;
+    }
+
+    @Override
+    public Municipality loadMunicipalityByCoordinates(double longitude,double latitude){
+        UriComponentsBuilder componentsBuilder = UriComponentsBuilder.fromHttpUrl("http://mapit.code4sa.org/point/4326/");
+        String latLong = longitude + "," + latitude;
+
+        componentsBuilder = componentsBuilder.path(latLong)
+                .queryParam("type","MN");
+
+        log.info("Composed URI: {}", componentsBuilder.toUriString());
+
+        ParameterizedTypeReference<Map<String,Municipality>> responseType =
+                new ParameterizedTypeReference<Map<String, Municipality>>() {};
+
+        ResponseEntity<Map<String,Municipality>> result =
+                restTemplate.exchange(componentsBuilder.build().toUri(),HttpMethod.GET,null,responseType);
+
+        Municipality municipality = null;
+        if(result.getBody() != null){
+            municipality = result.getBody().entrySet().iterator().next().getValue();
+        }https://www.youtube.com/watch?v=aKW3IWrPfaU&index=2&list=PLEvPW3QeeZUxRxbArpOOYKMi-zK3SV2x3
+
+        log.info("Municipality = {}",municipality);
+
+        return municipality;
+    }
+
+    @Override
+    public  List<UserLocationLog> loadUsersWithLocationNotNUll(Set<String> userUids){
+
+        List<UserLocationLog> userLocationLogs = userLocationLogRepository.findAll()
```

Comment:
Please stop doing this. I have said it many times now. Do not use find all from DB and then filter in memory. This is an anti-pattern and it is going to cause us massive problems. Don't do it. I can't say it many more times.

Do this as a DB query.